### PR TITLE
Document DEFAULT keyword as a reserved term

### PR DESCRIPTION
**Pull request by jhlodin**
## Description

`DEFAULT` is used as a keyword in Trino, at least in `SET PARAMETERS` clauses to revert a parameter back to its default value. It ought to be documented accordingly.

> Is this change a fix, improvement, new feature, refactoring, or other?

Improvement

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

Documentation

> How would you describe this change to a non-technical end user or system administrator?

Documenting the `DEFAULT` SQL keyword in Trino

## Related issues, pull requests, and links

* Fixes https://github.com/trinodb/trino/issues/10869
* Should be merged in conjunction with https://github.com/trinodb/trino/pull/11022
* Following up on https://github.com/trinodb/trino/pull/10835

## Documentation

( ) No documentation is needed.
(x) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(x) No release notes entries required.
( ) Release notes entries required with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Comment by jhlodin** [comment]
I considered adding a note about using `DEFAULT` in the `CREATE TABLE` and `CREATE TABLE AS` pages, to say that you can use it to set a default column value when defining a column. But:
* I don't know if we actually support that in Trino
* Maybe that's too much of a SQL fundamentals thing that it's overkill for us to document

Thoughts?

File: docs/src/main/sphinx/language/reserved.rst [hunk 1]
```diff
@@ -29,6 +29,7 @@ Keyword                        SQL:2016      SQL-92
 ``CURRENT_TIMESTAMP``          reserved      reserved
 ``CURRENT_USER``               reserved
 ``DEALLOCATE``                 reserved      reserved
+``DEFAULT``                    reserved      reserved
```
Comment (ebyhr) [inline docs/src/main/sphinx/language/reserved.rst:32]: `DEFAULT` isn't a reserved keyword in Trino at this time. That is why `docs-checks` job is failing.

**Comment by findepi** [comment]
cc @martint @kasiafi

File: docs/src/main/sphinx/language/reserved.rst (hunk 1 above)
Comment (jhlodin) [reply docs/src/main/sphinx/language/reserved.rst:32]: Maybe I'm misunderstanding, I thought that `DEFAULT` being an accepted keyword in, at least, the `... SET PROPERTIES foo = DEFAULT` context would mean that it's reserved?

Comment (kasiafi) [reply docs/src/main/sphinx/language/reserved.rst:32]: We generally try not to make keywords reserved unless there's a potential ambiguity. I don't think that's the case with `DEFAULT`. Also, making it reserved would likely cause backwards-compatibility issues.

**Comment by jhlodin** [comment]
Closing as inaccurate

**Comment by kasiafi** [comment]
> I considered adding a note about using DEFAULT in the CREATE TABLE and CREATE TABLE AS pages, to say that you can use it to set a default column value when defining a column. But:
I don't know if we actually support that in Trino

We don't support it.

**Pull request closed without merging**